### ISSUE TITLE
An infinite loop in ReadTXTImage():449, coders/txt.c

### ISSUE THREAD

**Issue by shqking** [ImageMagick/ImageMagick#713]
A mistake...

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by dlemstra** [comment]
@shqking Could you put your orginal message back? We did get your email but it is okay to post the message here. We are planning to add a contribution template that explains when you should contact us by e-maill.

**Comment by shqking** [comment]
Hi all,
Thanks for your prompt reply.
The details of this infinite loop issue are described as below.

We found an infinite loop bug in ReadTXTImage():449, coders/txt.c
The vulnerable code is shown as below.
```
447     image->columns=width;
448     image->rows=height;
449     for (depth=1; (GetQuantumRange(depth)+1) < max_value; depth++) ;
450     image->depth=depth;
```
An integer overflow might occur for the addition operation `GetQuantumRange(depth)+1` when **depth** is greater than 64, producing a smaller value than expected.
As a result, infinite loop would occur if a crafted TXT file, which claims very large **max_value**, i.e. a big value between `0x8000000000000001L` and `0xffff ffff ffff ffff`, is provided.

poc: [https://github.com/shqking/imagemagick-poc/blob/master/x_txt_poc.txt](url)
In our poc, we set **max_value** as `0x8000000000000001L`
The command we was using is `convert x_txt_poc.txt test.jpg`

Note that this issue is found by Xiaohei and Wangchu from Alibaba Security Team.
Thanks a lot.
